File: src/main/java/fr/vlegall/sochief/repository/IngredientUnitRepository.java
package fr.vlegall.sochief.repository;

import fr.vlegall.sochief.model.recipe.IngredientUnit;
import org.springframework.data.jpa.repository.JpaRepository;

public interface IngredientUnitRepository extends JpaRepository<IngredientUnit, Long> {
    IngredientUnit findByNameIgnoreCase(String name);
}

File: src/main/kotlin/fr/vlegall/sochief/repository/IngredientUnitRepository.kt
package fr.vlegall.sochief.repository

import fr.vlegall.sochief.model.recipe.IngredientUnit
import org.springframework.data.jpa.repository.JpaRepository

interface IngredientUnitRepository : JpaRepository<IngredientUnit, Long> {
    fun findByNameIgnoreCase(name: String): IngredientUnit?
}
